satori/resources/resource_data_access_common.go

```diff
@@ -77,6 +77,43 @@ func resourceDataAccessIdentity(isCelSupported bool) *schema.Schema {
 	}
 }
 
+func validateIdentityType(isCelSupported bool) func(interface{}, cty.Path) diag.Diagnostics {
+	return func(i interface{}, p cty.Path) diag.Diagnostics {
+
+		if i == "USER" ||
+			i == "DB_USER" ||
+			i == "GROUP" ||
+			i == "IDP_GROUP" ||
+			i == "DATABRICKS_GROUP" ||
+			i == "DATABRICKS_SERVICE_PRINCIPAL" ||
+			i == "SNOWFLAKE_ROLE" ||
+			i == "EVERYONE" ||
+			(isCelSupported && i == "CEL") {
+			return diag.Diagnostics{}
+		}
+
+		identityTypeErrMsg := fmt.Sprintf("Invalid identity type, `%s`,valid types are: %s", i, validIdentityTypeList(isCelSupported))
+
+		return diag.Diagnostics{
+			diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "Invalid value.",
+				Detail:   identityTypeErrMsg,
+			},
+		}
+	}
+}
+
+func validIdentityTypeList(isCelSupported bool) string {
+	identityTypeList := "USER, DB_USER, IDP_GROUP, GROUP, DATABRICKS_GROUP, DATABRICKS_SERVICE_PRINCIPAL, SNOWFLAKE_ROLE"
```

Comment:
Will the list appear in the documentation?

Reply:
ran `make docs`, then yes